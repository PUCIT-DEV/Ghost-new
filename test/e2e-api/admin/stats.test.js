const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyEtag, anyISODate} = matchers;

let agent;

describe('Stats API', function () {
    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('members');
        await agent.loginAsOwner();
    });

    it('Can fetch member count history', async function () {
        await agent
            .get(`/stats/member-count`)
            .expectStatus(200)
            .matchBodySnapshot({
                stats: [{
                    date: anyISODate
                }]
            })
            .matchHeaderSnapshot({
                etag: anyEtag
            });
    });
});
