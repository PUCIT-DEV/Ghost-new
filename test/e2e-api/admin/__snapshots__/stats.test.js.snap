// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats API Can fetch MRR history 1: [body] 1`] = `
Object {
  "meta": Object {
    "totals": Array [
      Object {
        "currency": "usd",
        "mrr": 1000,
      },
    ],
  },
  "stats": Array [
    Object {
      "currency": "usd",
      "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "mrr": 0,
    },
    Object {
      "currency": "usd",
      "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "mrr": 1000,
    },
  ],
}
`;

exports[`Stats API Can fetch MRR history 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "158",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Stats API Can fetch member count history 1: [body] 1`] = `
Object {
  "meta": Object {
    "totals": Object {
      "comped": 0,
      "free": 3,
      "paid": 5,
    },
  },
  "stats": Array [
    Object {
      "comped": 0,
      "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "free": 3,
      "paid": 5,
      "paid_canceled": 0,
      "paid_subscribed": 0,
    },
  ],
}
`;

exports[`Stats API Can fetch member count history 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "149",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Stats API Can fetch subscriptions history 1: [body] 1`] = `
Object {
  "meta": Object {
    "cadences": Array [
      "year",
      "month",
    ],
    "tiers": Array [
      StringMatching /\\[a-f0-9\\]\\{24\\}/,
    ],
    "totals": Array [
      Object {
        "cadence": "month",
        "count": 1,
        "tier": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      },
    ],
  },
  "stats": Array [
    Object {
      "cadence": "month",
      "count": 1,
      "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "negative_delta": 0,
      "positive_delta": 1,
      "tier": StringMatching /\\[a-f0-9\\]\\{24\\}/,
    },
    Object {
      "cadence": "year",
      "count": 0,
      "date": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "negative_delta": 0,
      "positive_delta": 0,
      "tier": StringMatching /\\[a-f0-9\\]\\{24\\}/,
    },
  ],
}
`;

exports[`Stats API Can fetch subscriptions history 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "403",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
